Sources/LeafKit/Data/TemplateData.swift

```diff
@@ -0,0 +1,242 @@
+import Foundation
+
+/// TemplateKit's supported serializable data types.
+/// - note: This is different from types supported in the AST.
+public struct TemplateData: Equatable, TemplateDataRepresentable {
+    // MARK: Equatable
+
+    /// See `Equatable`.
+    public static func ==(lhs: TemplateData, rhs: TemplateData) -> Bool {
+        /// Fuzzy compare
+        if lhs.string != nil && lhs.string == rhs.string {
+            return true
+        } else if lhs.int != nil && lhs.int == rhs.int {
+            return true
+        } else if lhs.double != nil && lhs.double == rhs.double {
+            return true
+        } else if lhs.bool != nil && lhs.bool == rhs.bool {
+            return true
+        }
+
+        /// Strict compare
```

Comment:
We should be able to get this for free if `TemplateDataStorage` conforms to `Equatable`